### PR TITLE
Feature/api endpoint create hot wallet address

### PR DESCRIPTION
**Pull request by swimsoft**
- Create endpoint to register hot wallet address with a project
- Fix offenses
- Generate new documentation
- Change authorization. update specs
- Fix specs
- Fix conflicts

### CODE REVIEW

File: app/controllers/api/v1/hot_wallet_addresses_controller.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    else
+      @errors = @build_hot_wallet.errors
+      render 'api/v1/error.json', status: :bad_request
+    end
+  end
+
+  private
+
+    def build_hot_wallet
+      @build_hot_wallet ||= project.create_hot_wallet(wallet_params)
```
Comment (aquabu) [inline app/controllers/api/v1/hot_wallet_addresses_controller.rb:26]: The method name build seems like it implies building the object in memory but not persisting it to the database? It seems like the `build_hot_wallet` and `project.create_hot_wallet` method names should match. They also should be named `build_not_wallet` if only in memory or `create_hot_wallet` when persisted to the database.

File: spec/controllers/api/v1/hot_wallet_addresses_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        params[:project_id] = project.id
+        params[:format] = :json
+
+        post :create, params: params
+        expect(response).to have_http_status(:created)
+        expect(project.reload.hot_wallet.present?).to be true
+      end
+    end
+
+    context 'when project has assign a hot wallet' do
```
Comment (aquabu) [inline spec/controllers/api/v1/hot_wallet_addresses_controller_spec.rb:43]: Minor typo:
'when project already has an assigned hot wallet'

**Review by aquabu — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/hot_wallet_addresses_controller.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  include Api::V1::Concerns::AuthorizableByMissionKey
+  include Api::V1::Concerns::RequiresAnAuthorization
+  include Api::V1::Concerns::RequiresSignature
+  include Api::V1::Concerns::RequiresWhitelabelMission
+
+  before_action :verify_hot_wallet, only: :create
+
+  def create
+    build_hot_wallet
+    @build_hot_wallet.source = Wallet.sources[:hot_wallet]
```
Comment (Riddlerrr) [inline app/controllers/api/v1/hot_wallet_addresses_controller.rb:11]: minor comment:
`@build_hot_wallet.source = :hot_wallet`?

Comment (swimsoft) [reply app/controllers/api/v1/hot_wallet_addresses_controller.rb:11]: 👍

Comment (swimsoft) [reply app/controllers/api/v1/hot_wallet_addresses_controller.rb:11]: 👍

File: app/controllers/api/v1/hot_wallet_addresses_controller.rb (hunk 1 above)
Comment (aquabu) [reply app/controllers/api/v1/hot_wallet_addresses_controller.rb]: @swimsoft I'm going to merge this but could you create a follow up commit that addresses this naming issue? Thanks.

**Review by aquabu — APPROVED** [review]

**Pull request merged**